stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/textinput/StreamMessageInputView.kt

```diff
@@ -0,0 +1,165 @@
+                        R.styleable.StreamMessageInputView_attachmentButtonIconPressedColor,
+                        ContextCompat.getColor(context, R.color.stream_gray_light)
+                    )
+                )
+            )
+
+            layoutParams.width = typedArray.getDimensionPixelSize(
+                R.styleable.StreamMessageInputView_attachmentButtonWidth,
+                context.resources.getDimensionPixelSize(R.dimen.stream_attachment_button_width)
+            )
+
+            layoutParams.height = typedArray.getDimensionPixelSize(
+                R.styleable.StreamMessageInputView_attachmentButtonHeight,
+                context.resources.getDimensionPixelSize(R.dimen.stream_attachment_button_height)
+            )
+        }
+    }
+
+    private fun configLightningButton(typedArray: TypedArray) {
+        binding.ivOpenEmojis.run {
+            isVisible =
+                typedArray.getBoolean(R.styleable.StreamMessageInputView_showLightningButton, true)
+
+            typedArray.getDrawable(R.styleable.StreamMessageInputView_lightningButtonIcon)
+                ?.let(this::setImageDrawable)
+
+            DrawableCompat.setTintList(
+                drawable,
+                getColorList(
+                    typedArray.getColor(
+                        R.styleable.StreamMessageInputView_lightningButtonIconColor,
+                        ContextCompat.getColor(context, R.color.stream_gray_dark)
+                    ),
+                    typedArray.getColor(
+                        R.styleable.StreamMessageInputView_lightningButtonIconPressedColor,
+                        ContextCompat.getColor(context, R.color.stream_white)
+                    ),
+                    typedArray.getColor(
+                        R.styleable.StreamMessageInputView_lightningButtonIconPressedColor,
+                        ContextCompat.getColor(context, R.color.stream_gray_light)
+                    )
+                )
+            )
+
+            layoutParams.width = typedArray.getDimensionPixelSize(
+                R.styleable.StreamMessageInputView_lightningButtonWidth,
+                context.resources.getDimensionPixelSize(R.dimen.stream_attachment_button_width)
+            )
+
+            layoutParams.height = typedArray.getDimensionPixelSize(
+                R.styleable.StreamMessageInputView_lightningButtonHeight,
+                context.resources.getDimensionPixelSize(R.dimen.stream_attachment_button_height)
+            )
+        }
+    }
+
+    private fun showSendMessageEnabled() {
+        binding.ivSendMessage.setImageResource(R.drawable.stream_ic_filled_up_arrow)
+    }
+
+    private fun hideSendMessageEnabled() {
+        binding.ivSendMessage.setImageResource(R.drawable.stream_ic_filled_right_arrow)
+    }
+
+    private fun configTextInput(typedArray: TypedArray) {
+        TextInputHandler(binding.etMessageTextInput, ::showSendMessageEnabled, ::hideSendMessageEnabled)
```

Comment:
🤔  There's no side effects, just the construction of the `TextWatcher`. The idea is that this class handles different behaviours related to text, like commands, reactions... I can get big, so it I would prefer to keep it out of the `StreamMessageInputView` so the second doesn't get super big.

Reply:
What I meant was that we're constructing a class instance, which executes code in its constructor (which is a side effect of constructing the instance), and then we never use the instance itself. We're creating a class instance to just run a block of code.

To simplify, what I'm trying to point out is that doing something like this:

```kotlin
class A(val param: Int) {
    init {
        println(param)
    }
}

A(x)
```

Is equivalent to a simple function call:

```kotlin
fun a(param: Int) {
     println(param)
}

a(x)
```

Reply:
If this `InputHandler` will eventually get larger, we can keep it around, but we could also move its `init` block into just a top level function, or maybe make it an `object` and call one of its functions, like so:

```kotlin
object A {
     fun init(param: Int) {
          println(param)
     }
}

A.init(param)
```

I just want to avoid using an `init` block and throwing away a class instance.

Reply:
Ok, makes sense. Removing the logic from the constructor